### PR TITLE
add asap3

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@jochym I see that you are on the ase package for conda-forge. Would you like to be on this one as well?

**Comment by CJ-Wright** [comment]
@conda-forge/core I think this is ready for review.

File: recipes/asap/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  url: https://gitlab.com/asap/asap/repository/archive.tar.gz?ref=ASAP_VERSION_3_10_0
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install --record record.txt
+  skip: True  # [not linux]
+
+requirements:
+  build:
```
Comment (jakirkham) [inline recipes/asap/meta.yaml]: Please add `toolchain`.

File: recipes/asap/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - python
+    - setuptools
+    - numpy
+    - ase
+  run:
+    - python
+    - numpy
```
Comment (jakirkham) [inline recipes/asap/meta.yaml]: If this links to `numpy`, it needs `numpy x.x`.

File: recipes/asap/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  number: 0
+  script: python setup.py install --record record.txt
+  skip: True  # [not linux]
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - numpy
+    - ase
```
Comment (jakirkham) [inline recipes/asap/meta.yaml]: Is this needed at build time or only at run time?

File: recipes/asap/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+source:
+  fn: asap-ASAP_VERSION_3_10_0-a78278070c997b302c8117d8174c8914d5f3d273.tar.gz
+  url: https://gitlab.com/asap/asap/repository/archive.tar.gz?ref=ASAP_VERSION_3_10_0
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install --record record.txt
+  skip: True  # [not linux]
```
Comment (jakirkham) [inline recipes/asap/meta.yaml]: Why only Linux?

File: recipes/asap/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
+{% set name = "asap" %}
+{% set version = "3.10.0" %}
+{% set sha256 = "d58105290029385835569e1e7057d7b03bad1dd0c5efc2151370e77f7721f478" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: asap-ASAP_VERSION_3_10_0-a78278070c997b302c8117d8174c8914d5f3d273.tar.gz
```
Comment (jakirkham) [inline recipes/asap/meta.yaml]: Really? Shouldn't. This only sets the name the file is saved under. The checksum should be invariant to that.

File: recipes/asap/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-ASAP_VERSION_{{ asap_version}}-a78278070c997b302c8117d8174c8914d5f3d273.tar.gz
+  url: https://gitlab.com/{{ name }}/{{ name }}/repository/archive.tar.gz?ref=ASAP_VERSION_{{ asap_version }}
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install --record record.txt
```
Comment (jakirkham) [inline recipes/asap/meta.yaml]: Also need `--single-version-externally-managed`.

File: recipes/asap/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    - python
+    - setuptools
+    - numpy x.x
+    - toolchain
+  run:
+    - python
+    - numpy x.x
+    - ase
+
+
```
Comment (jakirkham) [inline recipes/asap/meta.yaml]: nit: Please cut to 1 blank line.

File: recipes/asap/meta.yaml (hunk 6 above)
Comment (CJ-Wright) [inline recipes/asap/meta.yaml]: That seems to cause issues with the build. `error: option --single-version-externally-managed not recognized`

Comment (jakirkham) [inline recipes/asap/meta.yaml]: Ah ok, so you aren't using `setuptools` then. Please drop `--record record.txt`.

**Comment by CJ-Wright** [comment]
@jakirkham Do you want this squashed, assuming that you don't have any changes?

**Comment by jakirkham** [comment]
Not necessary.

Thanks @CJ-Wright .

File: recipes/asap/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+{% set version = "3.10.0" %}
+{% set sha256 = "d58105290029385835569e1e7057d7b03bad1dd0c5efc2151370e77f7721f478" %}
+{% set asap_version = "3_10_0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-ASAP_VERSION_{{ asap_version}}-a78278070c997b302c8117d8174c8914d5f3d273.tar.gz
```
Comment (jakirkham) [inline recipes/asap/meta.yaml]: Would try to simplify this further in the feedstock.

**Comment by CJ-Wright** [comment]
@jakirkham Thank you very much!

**Pull request merged**